README.md

```diff
@@ -149,6 +155,15 @@ to ensure there are no other PostgreSQL roles existing which are both superuser
 and can log in. Additionally there must be no unprivileged PostgreSQL roles
 which have been granted access to one of the existing superuser roles.
 
+#### Whitelist rules and logic
+
+For configuration options that end in `_whitelist` the following logic applies.
+
+* only roles explicitly listed or included by a group that is explicitly
+  listed (e.g. `'+admin'`) are allowed
+* if the whitelist is set to the empty set `''` all roles are denied
+* the default is `'*'` which means all roles are allowed
+
```

Comment:
I don't think the whitelist have different meanings in these cases, and I think being able to re-use the check_user_whitelist function illistrates that. The difference is in the target of the whitelist current_user vs the rolename argument and maybe wording/naming can fix that.

That said, I completely agree with you on a name change. My original hesitation on it was that I did not want to rename `set_user.superuser_whitelist` as I thought it make reading GUC variables confusing. I was unsure if the setting should reading from `set_user.superuser_source_whitelist` and then `set_user.superuser_whitelist`, in order to keep compatiable with current setups.

In summary, I will rename `set_user.dest_role_whitelist` to `set_user.nonsuperuser_target_whitelist` and `*Dest_role_whitelist` to `*Nonsuperuser_Target_Whitelist`, as those names have better meaning and no one is using them yet. However, I'll wait for you feedback on renaming  `set_user.superuser_whitelist` and how to handle reading from the guc for it's value. The other question is while I'm working on it would you like the C variables of `*SU_Whitelist` renamed, and if so what to?

Reply:
The re-use of `check_user_whitelist` is an implementation detail. Though it's nice that we can do that, it really has no bearing on the end-user that this doc is geared toward, IMO.

Good call re: `superuser_whitelist`. Maybe we leave that the same for now and move forward with the `dest_role` renaming. We may be able to do better than `nonsuperuser_target_whitelist`. Perhaps we should discuss the name out-of-band?